src/Utils/General.hs

```diff
@@ -4,6 +4,7 @@ Description : General utility functions
 
 module Utils.General where
 
-safeLast :: [a] -> Maybe a
-safeLast [] = Nothing
-safeLast xs = Just $ last xs
+-- unused
+--safeLast :: [a] -> Maybe a
+--safeLast [] = Nothing
+--safeLast xs = Just $ last xs
```

Comment:
This PR is already way too big so I won't ask for it here, but this is an empty file now. In the next PR we should remove this.

Reply:
I actually think it is worth keeping. I am certain we will have (if we do not already have) functions that are generally useful and should not be contained to a specific module. Keeping this file makes it more likely that a function like that will end up in the proper location rather than tucked away in another module. That is exactly the situation we had with the string utility functions, whose functionality was often being duplicated.